### PR TITLE
Remove bitcoincore-rpc dependency

### LINKED ISSUE #28: Replace `bitcoincore-rpc` with implementation based on `reqwest`

### PR DESCRIPTION
**Pull request by D4nte**
Resolves #28

### CODE REVIEW

File: tests/bitcoin_helper/mod.rs [hunk 1]
```diff
@@ -1,25 +1,291 @@
...
+#[derive(Debug)]
+pub enum Error {
+    Address(rust_bitcoin::util::address::Error),
+    Reqwest(reqwest::Error),
+    Encode(rust_bitcoin::consensus::encode::Error),
+    Rpc(RpcError),
+    Serialize(serde_json::Error),
+}
+
+impl From<reqwest::Error> for Error {
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply tests/bitcoin_helper/mod.rs]: https://docs.rs/thiserror/1.0.9/thiserror/

File: tests/rfc003_bitcoin_htlc.rs [hunk 2]
```diff
@@ -214,10 +205,9 @@ fn refund_htlc() {
         .send_raw_transaction(refund_tx_hex.clone())
         .unwrap_err();
 
-    // Can't access the type `RpcError`: https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/50
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply tests/rfc003_bitcoin_htlc.rs]: Means we should no longer match on the Debug impl of the struct but rather compare the messages directly!

File: tests/bitcoin_helper/mod.rs [hunk 3]
```diff
@@ -1,25 +1,291 @@
...
+        let request = JsonRpcRequest::<Vec<()>>::new("getnewaddress", Vec::new());
+
+        let response = reqwest::blocking::Client::new()
+            .post(self.endpoint.as_str())
+            .basic_auth(&self.auth.username, Some(&self.auth.password))
+            .json(&request)
+            .send()?
+            .json()?;
+
+        match response {
```
[the remark this answers is not in the document]
Comment (D4nte) [reply tests/bitcoin_helper/mod.rs]: There are actually 3 tests that rely on the error...

File: tests/bitcoin_helper/mod.rs [hunk 4]
```diff
@@ -1,25 +1,291 @@
...
+                result: Some(result),
+                error: None,
+            } => Ok(result),
+            _ => panic!("Received response with both result and error null"),
+        }
+    }
+
+    pub fn list_unspent(
+        &self,
+        minconf: Option<usize>,
```
[the remark this answers is not in the document]
Comment (D4nte) [reply tests/bitcoin_helper/mod.rs]: yes will remove

File: tests/bitcoin_helper/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply tests/bitcoin_helper/mod.rs]: But only for 1 particular rpc request right? I think for all the others, we want them to succeed for the test.

**Comment by D4nte** [comment]
> Also, please avoid merge commits as much as possible.

Please clarify your expectations as a rebase would mean that I need to `git push --force`.

**Comment by thomaseizinger** [comment]
> > Also, please avoid merge commits as much as possible.
> 
> Please clarify your expectations as a rebase would mean that I need to `git push --force`.

As long as there is no existing review, that is totally fine IMO.

**Comment by D4nte** [comment]
> Note: You'll need to rebase to get the new CI scripts I merged this morning.

This actually broke my CI run.... https://github.com/comit-network/blockchain-contracts/pull/34/checks?check_run_id=392512973

**Pull request merged**